### DISCUSSION TITLE
function get_load_average() does not exist

### DISCUSSION THREAD

**Discussion by LeDOC666** [cybertec-postgresql/pgwatch2#799] [Q&A]
Hello, I have a lot of error on my postgres databases monitored:

`2024-06-28 11:20:35.557 CEST [12068] pgwatch2@site_gpmm ERROR:  function get_load_average() does not exist at character 286
2024-06-28 11:20:35.557 CEST [12068] pgwatch2@site_gpmm HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
2024-06-28 11:20:35.557 CEST [12068] pgwatch2@site_gpmm STATEMENT:  SET lock_timeout TO '100ms';SET statement_timeout TO '5s';
        select
          (extract(epoch from now()) * 1e9)::int8 as epoch_ns,
          round(load_1min::numeric, 2)::float as load_1min,
          round(load_5min::numeric, 2)::float as load_5min,
          round(load_15min::numeric, 2)::float as load_15min
        from
          get_load_average();   -- needs the plpythonu proc from "metric_fetching_helpers" folder
`

After watching on different doc, I don't understand the problem:
My monitored database are version 12. 
I use pgwatch2 user (with pg_monitor role) to monitor the databases.
PGWatch2 is running in a Docker Compose, and watch all client databases (including my v12 one) with WEBUI configured.

Normally, with version 10+, and with pg_monitor role, I don't need to install anything on my postgres databases ?

**Comment by pashagolub** [comment]
Hello.

Have you followed the proper [preparation process](https://pgwatch2.readthedocs.io/en/latest/preparing_databases.html)? Seems like the helper function is missing

**Reply by LeDOC666** [reply]
Thanks for answer,

I've follow the "PL/Python helpers" step, but I had to create the function in the "dbname" database (db name filled in webui). May I create it in another one like postgres database ?
